test/unit/org/apache/cassandra/index/sai/cql/VectorUpdateDeleteTest.java

```diff
@@ -900,6 +900,30 @@ private void testBruteForceRangeQueryWithUpdatedVectors(int vectorDimension) thr
         });
     }
 
+    @Test
+    public void testVectorIndexWithAllOrdinalsDeletedViaRangeDeletion() throws Throwable
+    {
+        QueryController.QUERY_OPT_LEVEL = 0;
+        setMaxBruteForceRows(0);
+        createTable(KEYSPACE, "CREATE TABLE %s (pk int primary key, a int, str_val text, val vector<float, 3>)");
+        createIndex("CREATE CUSTOM INDEX ON %s(str_val) USING 'StorageAttachedIndex'");
+        createIndex("CREATE CUSTOM INDEX ON %s(val) USING 'StorageAttachedIndex'");
+        waitForIndexQueryable();
+        disableCompaction(KEYSPACE);
+
+        // Insert a row with a vector
+        execute("INSERT INTO %s (pk, a, str_val, val) VALUES (1, 1, 'A', [1.0, 2.0, 3.0])");
+
+        // Range delete that row
+        execute("DELETE FROM %s WHERE pk = 1");
+
+        // Insert another row without a vector
+        execute("INSERT INTO %s (pk, a, str_val) VALUES (2, 1, 'A')");
+        flush();
```

Comment:
We could certainly regress. Not an answer, but this is a weak spot in testing for C*, in my opinion. There are a many of optimizations where we could regress without a single test failure, and in this case, I don't think a regression would be too serious for us. However, I hate saying that because "now" is the best time to write a test on this kind of edge case. I looked into it, and it would be pretty hacky or would require a large amount of logic to test this case, so I'd prefer to skip it unless you think it's particularly important. Alternatively, we could assume that there are not any empty indexes out there and just remove the read path safety :)